### PR TITLE
Use parent class attribute  supports_out for foreach_zero opinfo

### PR DESCRIPTION
**Pull request by isuruf**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #112781
* __->__ #112778

Instead of introducing a new has_no_out_of_place attribute
Also fixes foreach_copy tests

### CODE REVIEW

File: test/test_foreach.py [hunk 1]
```diff
@@ -817,9 +817,9 @@ def test_foreach_copy_with_multi_device_inputs(self, device, dtype, op):
     def test_autodiff(self, device, dtype, op, inplace):
         if not (op.supports_autograd or op.supports_forward_ad):
             self.skipTest("neither reverse mode nor forward mode supported")
-        if (not inplace) and op.has_no_out_of_place:
+        if (not inplace) and not op.supports_out:
             self.skipTest("out-of-place not implemented")
-        if inplace and op.has_no_in_place:
+        if inplace and not op.has_no_in_place:
```
Comment (lezcano) [inline test/test_foreach.py:822]: Is this change correct? The name of the arg `has_no_in_place` is just not descriptive at all... ugh

**Review by lezcano — APPROVED** [review]
Just one comment, otherwise lgtm.

File: test/test_foreach.py (hunk 1 above)
Comment (isuruf) [reply test/test_foreach.py:822]: Reverted.

**Comment by lezcano** [comment]
this PR is already approved

**Pull request closed without merging**